### PR TITLE
Manual test feedbacks

### PR DESCRIPTION
**Pull request by mumoshu**
I'd like to add two fixes for the issues I've observed in manual testing. See each commit message for more details.

In nutshell, `RunnerDeployment` just works as explained in README after these fixes!

### CODE REVIEW

File: controllers/runnerset_controller.go [hunk 1]
```diff
@@ -164,3 +166,13 @@ func (r *RunnerSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+
+// workAroundRunnerSpecValidationBug "fixes" the runner spec by setting an empty "env" list to avoid
+// non-sense validation errors due to (I believe) a bug in controller-runtime CLient
+//
+// > validation failure list:\nspec.template.spec.env in body must be of type array: \"null\""
+func workAroundRunnerSpecValidationBug(spec *v1alpha1.RunnerSpec) {
+	if spec.Env == nil {
+		spec.Env = []corev1.EnvVar{}
+	}
+}
```
Comment (summerwind) [inline controllers/runnerset_controller.go]: Can you try adding 'omitempty' to 'Env' field to solve this problem?
https://github.com/summerwind/actions-runner-controller/blob/b9c6f463ed015f53dc762819b8a392634fde3cf0/api/v1alpha1/runner_types.go#L34

Comment (mumoshu) [reply controllers/runnerset_controller.go]: Thanks! That did solve my issue.

**Comment by mumoshu** [comment]
@summerwind Thanks for your review. This should be good to be reviewed once again :)

**Comment by summerwind** [comment]
LGTM 👍 Thank you!

**Pull request merged**